pkg/apis/apps/validation/validation_test.go

```diff
@@ -41,6 +50,28 @@ func TestValidateStatefulSet(t *testing.T) {
 			},
 		},
 	}
+	validPersistentVolumeClaimSpec := api.PersistentVolumeClaimSpec{
+		Selector: &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      "key2",
+					Operator: "Exists",
+				},
+			},
+		},
+		AccessModes: []api.PersistentVolumeAccessMode{
+			api.ReadWriteOnce,
+			api.ReadOnlyMany,
+		},
+		Resources: api.ResourceRequirements{
+			Requests: api.ResourceList{
+				api.ResourceName(api.ResourceStorage): resource.MustParse("10G"),
+			},
+		},
+		StorageClassName: &validClassName,
+	}
+
+	invalidClassName := "-invalid-"
```

Comment:
nit: 

"-invalid-" -> "invalid"?

Reply:
name should not consist of special character like "-" , so here construct one invalid name.

Reply:
ACK.

Reply:
`names must be a DNS subdomain`